pretyping/evarconv.ml

```diff
@@ -862,72 +923,84 @@ and evar_eqappr_x ?(rhs_is_already_stuck = false) ts env evd pbty
 
 	| Prod (n1,c1,c'1), Prod (n2,c2,c'2) when app_empty ->
             ise_and evd
-              [(fun i -> evar_conv_x ts env i CONV c1 c2);
+              [(fun i -> evar_conv_x flags env i CONV c1 c2);
                (fun i ->
  	         let c = nf_evar i c1 in
                  let na = Nameops.Name.pick n1 n2 in
-	         evar_conv_x ts (push_rel (RelDecl.LocalAssum (na,c)) env) i pbty c'1 c'2)]
+                 evar_conv_x flags (push_rel (RelDecl.LocalAssum (na,c)) env) i pbty c'1 c'2)]
 
 	| Rel x1, Rel x2 ->
 	    if Int.equal x1 x2 then
-              exact_ise_stack2 env evd (evar_conv_x ts) sk1 sk2
+              exact_ise_stack2 env evd (evar_conv_x flags) sk1 sk2
             else UnifFailure (evd,NotSameHead)
 
 	| Var var1, Var var2 ->
 	    if Id.equal var1 var2 then
-              exact_ise_stack2 env evd (evar_conv_x ts) sk1 sk2
+              exact_ise_stack2 env evd (evar_conv_x flags) sk1 sk2
             else UnifFailure (evd,NotSameHead)
 
 	| Const _, Const _
 	| Ind _, Ind _ 
 	| Construct _, Construct _ ->
-	  rigids env evd sk1 term1 sk2 term2
+           rigids env evd sk1 term1 sk2 term2
+
+        | Evar (sp1,al1), Evar (sp2,al2) -> (** Frozen evars *)
+          if Evar.equal sp1 sp2 then
+            match ise_stack2 false env evd (evar_conv_x flags) sk1 sk2 with
+            |None, Success i' ->
+              (** FIXME: solve_refl can restrict the evar, do we want to allow that? *)
+              Success (solve_refl flags (fun p env i pbty a1 a2 ->
```

Comment:
In general, when unifying X with X, pruning X is OK to me if the two explicit substitution do not unify.
But here it seems that X is frozen (rigid case) hence pruning should be forbidden.

I think the FIXME should be addressed.

Reply:
This is now adressed in solve_refl, indeed.